Compiler/src/abstractinterpretation.jl

```diff
@@ -50,35 +58,34 @@ function abstract_call_gf_by_type(interp::AbstractInterpreter, @nospecialize(f),
         return Future(CallMeta(Any, Any, Effects(), NoCallInfo()))
     end
 
-    (; valid_worlds, applicable, info) = matches
+    matches_boxed = SafeBox{Union{MethodMatches,UnionSplitMethodMatches}}(matches)
+    (; valid_worlds, applicable) = matches
     update_valid_age!(sv, valid_worlds)
 
     # final result
     gfresult = Future{CallMeta}()
     # intermediate work for computing gfresult
-    rettype = exctype = Bottom
-    conditionals = nothing # keeps refinement information of call argument types when the return type is boolean
-    seenall = true
-    const_results = nothing # or const_results::Vector{Union{Nothing,ConstResult}} if any const results are available
-    fargs = arginfo.fargs
-    all_effects = EFFECTS_TOTAL
-    slotrefinements = nothing # keeps refinement information on slot types obtained from call signature
+    rettype = SafeBox{Any}(Bottom)
+    exctype = SafeBox{Any}(Bottom)
+    conditionals_boxed = SafeBox{Union{Nothing,Tuple{Vector{Any},Vector{Any}}}}(nothing) # keeps refinement information of call argument types when the return type is boolean
+    seenall = SafeBox{Bool}(true)
+    const_results_boxed = SafeBox{Union{Nothing,Vector{Union{Nothing,ConstResult}}}}(nothing) # if any const results are available
+    all_effects = SafeBox{Effects}(EFFECTS_TOTAL)
```

Comment:
A single mutable struct might be more efficient here than the 7 separate allocations, if we want to work on trying to fine tune this

Reply:
Right. I’ve updated the implementation accordingly.